pkg/kubelet/cm/devicemanager/manager_test.go

```diff
@@ -1403,3 +1405,78 @@ func TestReadPreNUMACheckpoint(t *testing.T) {
 	err = m.readCheckpoint()
 	require.NoError(t, err)
 }
+
+func TestGetTopologyHintsWithUpdates(t *testing.T) {
+	socketDir, socketName, _, err := tmpSocketDir()
+	defer os.RemoveAll(socketDir)
+	require.NoError(t, err)
+
+	devs := []pluginapi.Device{}
+	for i := 0; i < 1000; i++ {
+		devs = append(devs, pluginapi.Device{
+			ID:     fmt.Sprintf("dev-%d", i),
+			Health: pluginapi.Healthy,
+			Topology: &pluginapi.TopologyInfo{
+				Nodes: []*pluginapi.NUMANode{
+					{ID: 0},
+				},
+			}})
+	}
+	topology := []cadvisorapi.Node{
+		{Id: 0},
+	}
+	testCases := []struct {
+		description string
+		count       int
+		devices     []pluginapi.Device
+		testfunc    func(manager *wrappedManagerImpl)
+	}{
+		{
+			description: "getAvailableDevices data race when update device",
+			count:       1,
+			devices:     devs,
+			testfunc: func(manager *wrappedManagerImpl) {
+				manager.getAvailableDevices(testResourceName)
+			},
+		},
+		{
+			description: "generateDeviceTopologyHints data race when update device",
+			count:       1,
+			devices:     devs,
+			testfunc: func(manager *wrappedManagerImpl) {
+				manager.generateDeviceTopologyHints(
+					testResourceName, sets.NewString(), sets.NewString(), 1)
+			},
+		},
+		{
+			description: "deviceHasTopologyAlignment data race when update device",
+			count:       1000,
+			devices:     devs[:1],
+			testfunc: func(manager *wrappedManagerImpl) {
+				manager.deviceHasTopologyAlignment(testResourceName)
+			},
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.description, func(t *testing.T) {
+			m, _ := setupDeviceManager(t, nil, nil, socketName, topology)
+			mimpl := m.(*wrappedManagerImpl)
+
+			updated := atomic.Bool{}
+			updated.Store(false)
```

Comment:
using waitgroups is more idiomatic https://gobyexample.com/waitgroups

Reply:
would it be more readable to define 2 go-routines and run them in parallel, e.g.
```
updated := atomic.Bool{}
updated.Store(false)

wg := sync.WaitGroup{}
wg.Add(2)

go func() {
    defer wg.Done()
    for i := 0; i < test.count; i++ {
        // simulate the device plugin to send device updates
	mimpl.genericDeviceUpdateCallback(testResourceName, devs)
    }
   updated.Store(true)
}()

go func() {
    defer wg.Done()
    for !updated.Load() {
        test.testfunc(mimpl)
    }
}()

wg.Wait()
```
?